assignment-3/src/presentation/Main.java

```diff
@@ -0,0 +1,132 @@
+package presentation;
+
+import data.ArrayListStudentDao;
+import data.StudentDao;
+import logic.Student;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+public class Main {
+    private static final String ADD_MENU_ITEM = "1";
+    private static final String UPDATE_MENU_ITEM = "2";
+    private static final String LIST_MENU_ITEM = "3";
+    private static final String DELETE_MENU_ITEM = "4";
+
+    private static final String EXIT_MENU_ITEM = "6";
+    private static final String NO_MENU_ITEM = "0";
+    public static final String SORT_AND_SHOW = "5";
+    public static final int SORT_BY_ID = 1;
+    public static final int SORT_BY_FIRST_NAME = 2;
+    public static final int SORT_BY_LAST_NAME = 3;
+
+
+    public static void main(String[] args) {
+        String choice = NO_MENU_ITEM;
+        StudentDao studentDao = new ArrayListStudentDao();
+        Scanner scanner = new Scanner(System.in);
+
+        while (!choice.equals(EXIT_MENU_ITEM)) {
+            System.out.println("Here are the available menu items:");
+            System.out.println("1. Add student");
+            System.out.println("2. Update student");
+            System.out.println("3. Show list of students");
+            System.out.println("4. Delete student");
+            System.out.println("5. Sort and Show");
+            System.out.println("6. Exit");
+            System.out.println("___________________________");
+
+            System.out.print("Please enter your choice:");
+            choice = scanner.nextLine();
+            System.out.println();
+
+            switch (choice) {
+                case ADD_MENU_ITEM -> {
+                    Student newStudent = new Student();
+                    System.out.println("Student ID:");
+                    newStudent.setId(scanner.nextInt());
+                    scanner.nextLine();
+                    System.out.print("First name:");
+                    newStudent.setFirstName(scanner.nextLine());
+                    System.out.print("Last name:");
+                    newStudent.setLastName(scanner.nextLine());
+                    try {
+                        studentDao.add(newStudent);
+                    } catch (Exception e) {
```

Comment:
Here you should catch the custom `StudentAlreadyExistsException` and output some message to the user instead of throwing a `RuntimeException`.